src/reconcile/rec_visibility.c

```diff
@@ -296,8 +298,10 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
          * check is not required for history store updates as they are implicitly committed. As
          * prepared transaction IDs are globally visible, need to check the update state as well.
          */
-        if (!is_hs_page && (F_ISSET(r, WT_REC_VISIBLE_ALL) ? WT_TXNID_LE(r->last_running, txnid) :
-                                                             !__txn_visible_id(session, txnid))) {
+        if ((F_ISSET(r, WT_REC_EVICT) && txn_global->checkpoint_running && is_hs_page &&
+              WT_TXNID_LT(txn_global->checkpoint_id, txnid)) ||
```

Comment:
Is it possible checkpoint starts to run in the middle of the eviction and the data should not be evicted in the first half of the page is evicted and the second half not?

Reply:
This also doesn't account for snapshot isolation and some transactions that may have less transaction ids than the checkpoint but still should not be visible.

Reply:
I agree its not the final solution however I'm not sure how I could account for checkpoints snapshot here.